src/dos/dos_devices.cpp

```diff
@@ -175,7 +157,7 @@ void DOS_DelDevice(DOS_Device * dev) {
 // We will destroy the device if we find it in our list.
 // TODO:The file table is not checked to see the device is opened somewhere!
 	for (Bitu i = 0; i <DOS_DEVICES;i++) {
-		if(Devices[i] && !strcasecmp(Devices[i]->name,dev->name)){
+		if (Devices[i] && !strcasecmp(Devices[i]->GetName(), dev->GetName())) {
```

Comment:
With the underlying `DOS_File` now free of pointers, it's comparison `operator ==` will do what we would expect. So in theory we could do `if (Devices[i] && Devices[i] == dev)`, however that also compares the other members (`attr`, `open`, `refctr`) in addition to the `name`.. which would probably return false-negatives.

Reply:
The places in code I looked at all could've compared by pointer and not by name, but I don't know - maybe I missed something. I don't want to touch it in case it leads to a bigger refactor and the code seems fine as it is right now.